stages/org.osbuild.ostree.deploy-verity

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/python3
+"""Enable fs-verity on an ostree repo/deployment
+
+The way osbuild works the ostree deployment is built in a chroot and
+stored as a regular directory of files before finally being copied to
+the physical filesystem. This means ostree fs-verity support doesn't
+work, as the fs-verity setting of files is not copied.
+
+To support fs-verity in generated images you have to run this stage
+after copying the final ostree tree onto the target filesystem.
+
+Note: Ensure the target filesystem supports fs-verity. See e.g.  the
+`verity` option in org.ostbuild.mkfs.ext4.
+
+"""
+
+import fcntl
+import os
+import struct
+import sys
+
+import osbuild.api
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "required": ["repo"],
+  "properties": {
+     "repo": {
+      "description": "Location of the OSTree repo.",
+      "type": "string"
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true
+},
+"mounts": {
+  "type": "array"
+}
+"""
+
+FS_IOC_ENABLE_VERITY = 0x40806685
+FS_VERITY_HASH_ALG_SHA256 = 1
+
+
+def enable_fsverity(path):
+    with open(path, "rb") as f:
+        try:
+            fcntl.ioctl(f, FS_IOC_ENABLE_VERITY, struct.pack("IIIIQIIQ11Q",
+                                                             1,
+                                                             FS_VERITY_HASH_ALG_SHA256,
+                                                             4096,
+                                                             0, 0,
+                                                             0, 0, 0,
+                                                             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0))
+        except FileExistsError:
+            pass  # fsverity already enabled
+
+
+def ostree_deploy_enable_verity(deploydir):
+    cfsfile = os.path.join(deploydir, ".ostree.cfs")
```

Comment:
IMO this is hardcoding an implementation detail; enabling fsverity on the objects should be owned by ostree.

Reply:
Ughhh....

Well, one argument here is that `org.osbuild.copy` should propagate the fsverity state to start.  Oh I see, it's just a thin wrapper around `cp -a` which would have to change.

So...I think what we may want here instead is something like `ostree admin deploy --post-copy` or something that does any necessary fixups after an ostree system root is copied like this.

Reply:
> Well, one argument here is that `org.osbuild.copy` should propagate the fsverity state to start. Oh I see, it's just a thin wrapper around `cp -a` which would have to change.

Its easy to guarantee that the final target supports fs-verity, because the osbuild manifests picks fs type and runs the mkfs. However, we can't guarantee that the intermediate storage that osbuild uses supports fs-verity, so in fact there may not be anything to preserve.
 
> So...I think what we may want here instead is something like `ostree admin deploy --post-copy` or something that does any necessary fixups after an ostree system root is copied like this.

Yeah, that sounds like a reasonable approach. It could even read the fs-verity option in the config. I'll have a go at adding that.